jakarta-package-alignment/src/test/groovy/com/palantir/gradle/jakartapackagealignment/JakartaPackageAlignmentPluginIntegrationSpec.groovy

```diff
@@ -16,13 +16,99 @@
 
 package com.palantir.gradle.jakartapackagealignment
 
+import groovy.transform.CompileStatic
 import nebula.test.IntegrationSpec
+import nebula.test.dependencies.DependencyGraph
+import nebula.test.dependencies.GradleDependencyGenerator
 
 /**
  * Integration tests for {@link JakartaPackageAlignmentPlugin}.
  */
 class JakartaPackageAlignmentPluginIntegrationSpec extends IntegrationSpec {
 
-    // TODO
+    static def PLUGIN_NAME = "com.palantir.jakarta-package-alignment"
+    private File mavenRepo
+    private File versionsProps
 
+    // borrowed from gcv
+    @CompileStatic
+    protected File generateMavenRepo(String... graph) {
+        DependencyGraph dependencyGraph = new DependencyGraph(graph)
+        GradleDependencyGenerator generator = new GradleDependencyGenerator(
+                dependencyGraph, new File(projectDir, "build/testrepogen").toString())
+        return generator.generateTestMavenRepo()
+    }
+
+    void setup() {
+        // TODO: this could potentially be replaced by something that calls VersionMappings.getReplacement
+        // at runtime to keep the jakarta and javax versions consistent, but we'd still need to hardcode versions
+        // somewhere in this test code anyway...
```

Comment:
:+1: I think this is fine as is. Most important thing is having a template where we can repro the issues that come up in practice :-)